src/mmtk.rs

```diff
@@ -596,4 +597,14 @@ impl<VM: VMBinding> MMTK<VM> {
             .vm_space
             .initialize_object_metadata(object, false)
     }
+
+    pub fn inspect_spaces<'a>(&'a self) -> Vec<&'a dyn SpaceInspector> {
```

Comment:
Yes.  Having a `Vec` does simplify the API a bit because it can be iterated, and the performance doesn't matter because (1) we are inspecting, and (2) there are probably not may spaces in a plan.